trpcultivate_phenoshare/src/Plugin/TripalImporter/TripalCultivatePhenoshareImporter.php

```diff
@@ -56,18 +56,31 @@ class TripalCultivatePhenoshareImporter extends ChadoImporterBase {
     'Header 2' => 'Header 2 Description',
     'Header 3' => 'Header 3 Description',
   ];
+  
+  // Service: Make the following services available to all stages.
+  // Genus Of Project service.
+  protected $service_genusproject;
+  // Genus Ontology configuration service.
+  protected $service_genusontology;
 
   /**
    * {@inheritDoc}
    */
   public function form($form, &$form_state) {
+    // Set genus project service.
+    $this->service_genusproject  = \Drupal::service('trpcultivate_phenotypes.genus_project');
```

Comment:
You'll want to add these via dependancy injection. Here is a tutorial focused on how to do this in forms: https://www.drupal.org/docs/drupal-apis/services-and-dependency-injection/dependency-injection-for-a-form

Reply:
I get an error, to me, conflicts with the create method of the ChadoImporterBase???

Fatal error: Declaration of Drupal\trpcultivate_phenoshare\Plugin\TripalImporter\TripalCultivatePhenoshareImporter::create(Symfony\Component\DependencyInjection\ContainerInterface $container) must be compatible with Drupal\tripal_chado\TripalImporter\ChadoImporterBase::create(Symfony\Component\DependencyInjection\ContainerInterface $container, array $configuration, $plugin_id, $plugin_definition)

This is very very interesting resource but we don't want to do this, maybe?
https://www.phptutorial.net/php-oop/php-override-method/

Reply:
Based on the error,
- Method in ChadoImporterBase has $container, $configuration, $plugin_id, $plugin_definition
- Method in your importer had only $container.

To solve this you should just have needed to match the parameters used in create in the chadoimporterbase.

Reply:
an override, will try

Reply:
I could not get the constructor to stop complaining about mismatch parameter despite providing the exact same parameter type and count. I found other method of injecting service through the user of setter method... works! manual test is good and all tests passing.

Setter injection
https://medium.com/luciushq/dependency-injection-in-drupal-8-an-introduction-8b3bdf044932